packages/reactivity-transform/src/reactivityTransform.ts

```diff
@@ -550,6 +550,58 @@ export function transformAST(
     }
   }
 
+  /**
+   * unwrap the code form the macro($、$$), fix #6312 and keep the ideally behavior with the RFC#369
+   */
+  function unwrapMacro(node: CallExpression) {
+    const argsLength = node.arguments.length
+    const bracketStart = node.callee.end! + offset
+
+    s.remove(node.callee.start! + offset, bracketStart)
+
+    if (argsLength === 1) {
+      const bracketEnd = node.arguments[argsLength - 1].end! + offset
+
+      // remove brackets of macros
+      s.remove(bracketStart, bracketStart + 1)
+      s.remove(bracketEnd, bracketEnd + 1)
+
+      // avoid traversal of like `$$(a)`
+      if(node.arguments[0].type!=="CallExpression"){
+        return;
+      }
+    }
+
+    // avoid invalid traversal for $
+    if (!escapeScope|| node.extra) {
+      return
+    }
+
+    // fix some edge cases for macro $$, eg. $$(a,b)
+    // resolve nested
+    let i = parentStack.length - 1
+    while (i >= 0) {
+      const curParent = parentStack[i--]
+      if (
+        curParent.type === 'ExpressionStatement' &&
+        isIsolateExpression(curParent.expression, node)
+      ) {
+        // split the unwrapped code `()()` to `();()`
+        s.appendLeft(node.callee.start! + offset, ';')
+        return
+      }
+    }
+  }
+
+  function isIsolateExpression(expression: Expression, node: CallExpression) {
```

Comment:
we could move this function to the top level.

Reply:
too tired 😭